diff --git a/activerecord/test/schema/schema.rb b/activerecord/test/schema/schema.rb
index 8a3dfbb..3dfa18c 100644
--- a/activerecord/test/schema/schema.rb
+++ b/activerecord/test/schema/schema.rb
@@ -746,7 +746,7 @@ ActiveRecord::Schema.define do
     t.string 'a$b'
   end
 
-  except 'SQLite' do
+  except ['SQLite', 'Akiban'] do
     # fk_test_has_fk should be before fk_test_has_pk
     create_table :fk_test_has_fk, :force => true do |t|
       t.integer :fk_id, :null => false
